src/napistu/mechanism_matching.py

```diff
@@ -1,28 +1,121 @@
 from __future__ import annotations
 
 import logging
+from typing import Optional, Union, Set, Dict, List
 
 import igraph as ig
+import numpy as np
 import pandas as pd
+
+from napistu import identifiers
 from napistu import sbml_dfs_core
 from napistu import utils
 from napistu.constants import SBML_DFS
 from napistu.constants import CPR_EDGELIST
 from napistu.constants import CPR_EDGELIST_REQ_VARS
+from napistu.constants import FEATURE_ID_VAR_DEFAULT
+from napistu.constants import RESOLVE_MATCHES_AGGREGATORS
+from napistu.constants import RESOLVE_MATCHES_TMP_WEIGHT_COL
 from napistu.constants import IDENTIFIERS
 from napistu.constants import IDENTIFIER_EDGELIST_REQ_VARS
-from napistu.constants import SPECIES_IDENTIFIERS_REQUIRED_VARS
+from napistu.constants import ONTOLOGIES_LIST
 from napistu.network.constants import CPR_GRAPH_EDGES
 from napistu.network import paths
 
 logger = logging.getLogger(__name__)
 
 
+def bind_wide_results(
+    sbml_dfs : sbml_dfs_core.SBML_dfs,
+    results_df : pd.DataFrame,
+    results_name : str,
+    ontologies : Optional[Union[Set[str], Dict[str, str]]] = None,
+    dogmatic : bool = False,
+    species_identifiers : Optional[pd.DataFrame] = None,
+    feature_id_var : str = FEATURE_ID_VAR_DEFAULT,
+    numeric_agg : str = RESOLVE_MATCHES_AGGREGATORS.WEIGHTED_MEAN,
+    keep_id_col : bool = True,
+    verbose : bool = False
+) -> sbml_dfs_core.SBML_dfs:
+    """
+    Binds wide results to a sbml_dfs object.
+
+    Take a table with molecular species-level attributes tied to systematic identifiers and match them to an sbml_dfs_model transferring these attributes to species_data
+
+    Parameters
+    ----------
+    sbml_dfs : sbml_dfs_core.SBML_dfs
+        The sbml_dfs object to bind the results to.
+    results_df : pd.DataFrame
+        The table containing the results to bind.
+    results_name : str
+        The name of the results to bind.
+    ontologies : Optional[Union[Set[str], Dict[str, str]]]
```

Comment:
To me it is not clear what `Dict[str, str]` will do

Reply:
Maybe add explaination from `match_by_ontology_and_identifier`:
"    ontologies : Optional[Union[Set[str], Dict[str, str]]], default=None
        Either:
        - Set of columns to treat as ontologies
        - Dict mapping wide column names to ontology names
        - None to automatically detect ontology columns based on ONTOLOGIES_LIST"